Release/src/http/client/http_client_asio.cpp

```diff
@@ -1091,6 +1088,7 @@ class asio_context final : public request_context, public std::enable_shared_fro
         }
         else
         {
+            m_timer.start_communicating();
```

Comment:
This is inside the `write_request()` that starts sending the request to the server. Just before the respective `async_write()` we call `m_timer.start_communicating()` to switch the `timeout_timer` state from the "connection timeout" to the "global timeout".

Note that this method might also initiate a TLS handshake if required (see conditional above). In that case, the request is eventually sent inside `handle_handshake()` and _not_ here.